client/scripts/views/pages/discussions/index.ts

```diff
@@ -67,41 +66,85 @@ const DiscussionsPage: m.Component<{ topic?: string }, IDiscussionPageState> = {
       app.user.unseenPosts[app.activeId()]['activePosts'] = 0;
       app.user.unseenPosts[app.activeId()]['threads'] = 0;
     }
-
-    // Infinite Scroll
-    const onscroll = _.debounce(() => {
-      const scrollHeight = $(document).height();
-      const scrollPos = $(window).height() + $(window).scrollTop();
-      if (scrollPos > (scrollHeight - 400)) {
-        if (!vnode.state.postsDepleted) {
-          vnode.state.lookback += vnode.state.defaultLookback;
-          m.redraw();
-        }
-      }
-    }, 400);
-    $(window).on('scroll', onscroll);
   },
   oninit: (vnode) => {
-    // determine lookback length
-    vnode.state.defaultLookback = 20;
-
+    vnode.state.lookback = {};
+    // TODO: This will become a problem if anyone creates a topic named allProposals
+    const subpage = vnode.attrs.topic || 'allProposals';
     const returningFromThread = (app.lastNavigatedBack() && app.lastNavigatedFrom().includes('/proposal/discussion/'));
-    vnode.state.lookback = (returningFromThread && localStorage[`${app.activeId()}-lookback`])
-      ? Number(localStorage[`${app.activeId()}-lookback`])
-      : Number.isInteger(vnode.state.lookback)
-        ? vnode.state.lookback
-        : vnode.state.defaultLookback;
+    vnode.state.lookback[subpage] = (returningFromThread && localStorage[`${app.activeId()}-lookback-${subpage}`])
+      ? localStorage[`${app.activeId()}-lookback-${subpage}`]
+      : vnode.state.lookback[subpage]?._isAMomentObject
+        ? vnode.state.lookback[subpage]
+        : moment();
+    vnode.state.topicInitialized = {};
   },
   view: (vnode) => {
     const { topic } = vnode.attrs;
     const activeEntity = app.community ? app.community : app.chain;
+    const subpage = topic || 'allProposals';
```

Comment:
we could use something less likely to be a topic id, such as "COMMONWEALTH_ALL_PROPOSALS". also we should define this constant once somewhere and import it from there

Reply:
That sounds good to me.